packages/core/src/validate/run.ts

```diff
@@ -46,6 +52,29 @@ interface ValidationErrorOpcode extends ValidationErrorBase {
   kind: 'delegatecall' | 'selfdestruct';
 }
 
+interface ValidationErrorUpgradeability extends ValidationErrorBase {
+  kind: 'no-public-upgrade-fn';
+}
+
+function skipCheck<T>(error: string, node: T): boolean {
```

Comment:
`node` should have type `Node`. (`import { Node } from 'solidity-ast/node'`)

This is the type of an arbitrary node in the AST. There is a problem, though, in that not every node has a `documentation` field, so instead of casting to `any` you should use `if ('documentation' in node)`.

I'm trying to understand why you implemented this function in this way with an internal function and a recursive call. Are we using any contracts compiled with Solidity 0.5? I feel like we only support Solidity 0.6. It seems 0.5 is in the Hardhat config but I'm not sure why... I'm asking this because if I recall correctly from 0.6 onwards the `documentation` field is always an object with a `text` field ([`StructuredDocumetation`](https://solidity-ast.netlify.app/interfaces/structureddocumentation)).

Reply:
The main issue here is that solidity-ast doesn't support documentation everywhere. I opened a PR in the soldity ast repo that include a contract that compiles, but is not parsed correctly. Not sure we can do the validation/run part cleanly unless solidity-ast is fixed, and I haven't been able to fix it (I've tried)

Reply:
I've updated it to avoid the any cast, but I still cannot assume node.documentation is null | StructuredDocumentation ... I still get strings :/